src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventLogger.java

```diff
@@ -168,7 +168,7 @@ private void handleTick() {
   }
 
   private void maybeRolloverWriterForDay() throws IOException {
```

Comment:
The method can be called from different threads. I would recommend handling synchronization in some way.

Reply:
Now `maybeRolloverWriterForDay` is called in one place, the same as `createWriter`.  Do you think it might be better to introduce syncronization point at the place of invocation instead, e.g.:

```
synchronized (this) {
  writer = recordsWriterFactory.createWriter();
  recordsWriterFactory.maybeUpdateRolloverTime();
}
```

Reply:
The method `maybeRolloverWriterForDay` is called in 2 places: `handleTick` and `writeEventWithRetries`.

If you've meant `maybeUpdateRolloverTime`, then the synchronization needs to enclose both calls: `shouldRollover()` and `maybeUpdateRolloverTime()`.

Not only that. All variables that can change and that are accessed from mutliple threads need to be synchronized:
- writer - not only in this method, but also in writeEventWithRetries
- fields inside DatePartitionedRecordsWriterFactory

The easiest approach would be to synchronize writeEventWithRetries and handleTick.

Reply:
I made `writeEventWithRetries` and `handleTick` syncronized